autoload/go/util.vim

```diff
@@ -49,6 +49,17 @@ function! go#util#IsMac() abort
         \ go#util#Exec(['uname'])[0] =~? '^darwin'
 endfunction
 
+" IsCygwin returns 1 if current OS is Cygwin/MSYS2/GitBash, 0 otherwise
+function! go#util#IsCygwin()
+  return !has('win32') &&
+        \ has('win32unix') &&
+        \ (
+        \ system('uname') =~ 'CYGWIN' ||
+        \ system('uname') =~ 'MINGW' ||
+        \ system('uname') =~ 'MSYS'
+        \ )
```

Comment:
Please remove the `uname` checks. They are unnecessary.

There's also no need to check for `!has('win32')`, as you can see in the matrix you provided; `win32` and `win32unix` are mutually exclusive. Please remove that check.

Reply:
Done. Thanks !